app/assets/javascripts/comfortable_mexican_sofa/admin/modules/EmbeddedGoogleAnalytics.js

```diff
@@ -0,0 +1,108 @@
+define(['jquery', 'DoughBaseComponent', 'eventsWithPromises', 'async!https://apis.google.com/js/platform.js!onload'], function($, DoughBaseComponent, eventsWithPromises) {
+  'use strict';
+
+  var defaultConfig = { };
+
+  function EmbeddedGoogleAnalytics($el, config) {
+    EmbeddedGoogleAnalytics.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   */
+  DoughBaseComponent.extend(EmbeddedGoogleAnalytics);
+
+  EmbeddedGoogleAnalytics.componentName = 'EmbeddedGoogleAnalytics';
+
+  EmbeddedGoogleAnalytics.prototype.init = function(initialised) {
+    gapi.analytics = {
+      q:[],
+      ready:function(f){this.q.push(f);}
+    };
+
+    gapi.analytics.ready($.proxy(function() {
+      this.onReady.call(this);
+    }, this));
+
+    gapi.load('analytics');
+
+    this._initialisedSuccess(initialised);
+
+    return this;
+  };
+
+  EmbeddedGoogleAnalytics.prototype.onReady = function() {
+    var path = this.config.path;
+
+    gapi.analytics.auth.authorize({
+      container: 'embed-api-auth-container',
+      clientid: '46744299348-7t8tn92mh4pv01vk89llfv0u3q7kup2j.apps.googleusercontent.com',
+    });
+
+    var dataChart1 = this.createChart({ "path": path, "metrics": "ga:pageviews", "container": "chart-1-container" });
+    var dataChart2 = this.createChart({ "path": path, "metrics": "ga:uniquePageviews", "container": "chart-2-container" });
+    var dataChart3 = this.createChart({ "path": path, "metrics": "ga:avgTimeOnPage", "container": "chart-3-container" });
+    var dataChart4 = this.createChart({ "path": path, "metrics": "ga:exitRate", "container": "chart-4-container" });
+    var dataChart5 = this.createChart({ "path": path, "metrics": "ga:bounceRate", "container": "chart-5-container" });
+
+    this.viewSelector1 = new gapi.analytics.ViewSelector({
```

Comment:
Cool, perhaps the `createChart` method could also set up the ViewSelector, execute, and event listeners too? You have all the context you need in those functions. It could dynamically set the instance variables too:

```js
this.viewSelectors = {};
```

then in the function

```js
this.viewSelectors[viewContainer] = new gapi.analytics.ViewSelector({
  container: viewContainer
});
```

Just a suggestion to DRY it up a little...